src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -0,0 +1,503 @@
+                var jobs = act.Jobs.OrderBy(job =>
+                {
+                    switch (job.VariantCase)
+                    {
+                        case WorkflowActivationJob.VariantOneofCase.NotifyHasPatch:
+                            return 0;
+                        case WorkflowActivationJob.VariantOneofCase.SignalWorkflow:
+                            return 1;
+                        case WorkflowActivationJob.VariantOneofCase.QueryWorkflow:
+                            return 3;
+                        default:
+                            return 2;
+                    }
+                });
+
+                // Run the event loop until yielded for each job
+                try
+                {
+                    var previousContext = SynchronizationContext.Current;
+                    try
+                    {
+                        // We must set the sync context to null so work isn't posted there
+                        SynchronizationContext.SetSynchronizationContext(null);
+                        foreach (var job in jobs)
+                        {
+                            Apply(job);
+                            // Run scheduler once. Do not check conditions when patching or querying.
+                            var checkConditions = job.NotifyHasPatch == null && job.QueryWorkflow == null;
+                            RunOnce(checkConditions);
+                        }
+                    }
+                    finally
+                    {
+                        SynchronizationContext.SetSynchronizationContext(previousContext);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Logger.LogWarning(
+                        e,
+                        "Failed activation on workflow {WorkflowType} with ID {WorkflowID} and run ID {RunID}",
+                        Info.WorkflowType,
+                        Info.WorkflowID,
+                        Info.RunID);
+                    try
+                    {
+                        completion.Failed = new()
+                        {
+                            Failure_ = details.FailureConverter.ToFailure(e, details.PayloadConverter),
+                        };
+                    }
+                    catch (Exception inner)
+                    {
+                        Logger.LogError(
+                            inner,
+                            "Failed converting activation exception on workflow with run ID {RunID}",
+                            Info.RunID);
+                        completion.Failed = new()
+                        {
+                            Failure_ = new() { Message = $"Failed converting activation exception: {inner}" },
+                        };
+                    }
+                }
+
+                // Remove any non-query commands after terminal commands
+                if (completion.Successful != null)
+                {
```

Comment:
You shouldn't have to do this - why don't you open an issue in core and I can handle this there and then we can get rid of this code.

Reply:
Opened https://github.com/temporalio/sdk-core/issues/497